testing/src/test/java/org/oppia/android/testing/data/AsyncResultSubjectTest.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.testing.data
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.util.data.AsyncResult
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class AsyncResultSubjectTest {
+
+  @Test
+  fun testAsyncResultPending_isPending() {
+    val pendingResult: AsyncResult<String> = AsyncResult.Pending()
+
+    AsyncResultSubject.assertThat(pendingResult).isPending()
+  }
+
+  @Test
+  fun testAsyncResultPending_isNotSuccess() {
+    val pendingResult: AsyncResult<String> = AsyncResult.Pending()
+
+    AsyncResultSubject.assertThat(pendingResult).isNotSuccess()
+  }
+
+  @Test
+  fun testAsyncResultSuccess_isSuccess() {
+    val successResult: AsyncResult<String> = AsyncResult.Success("Success value")
+
+    AsyncResultSubject.assertThat(successResult).isSuccess()
+  }
+
+  @Test
+  fun testAsyncResultSuccess_hasSuccessValueWhere_matchesExpected() {
+    val successResult: AsyncResult<String> = AsyncResult.Success("Success value")
+
+    AsyncResultSubject.assertThat(successResult).hasSuccessValueWhere {
+      // Here we are verifying that the value is "Success value"
+      assertThat(this).isEqualTo("Success value")
+    }
+  }
```

Comment:
Suggestion to reduce ambiguity.
```suggestion
  @Test
  fun testAsyncResultSuccess_hasSuccessValueWhere_matchesExpected() {
    val successResult: AsyncResult<String> = AsyncResult.Success("Some string")

    AsyncResultSubject.assertThat(successResult).hasSuccessValueWhere { result ->
      assertThat(result).isEqualTo("Some string")
    }
  }
```